src/Avalonia.Build.Tasks/XamlCompilerDiagnosticsFilter.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using XamlX;
+
+namespace Avalonia.Build.Tasks;
+
+public class XamlCompilerDiagnosticsFilter
+{
+    private static readonly Regex s_editorConfigRegex =
+        new("""avalonia_xaml_diagnostic\.([\w\d]+)\.severity\s*=\s*(\w*)""");
+
+    private readonly bool _treatWarningsAsErrors;
+    private readonly HashSet<string> _warningsAsErrors;
+    private readonly HashSet<string> _warningsNotAsErrors;
+    private readonly HashSet<string> _noWarn;
```

Comment:
It works a bit weird, but it seems like we don't need to explicitly handle these properties. And MSBuild will automatically treat warnings as it decides, assuming a diagnostic code is matching.

But still, we need to manually check for editor config, as MSBuild doesn't do it.

Reply:
todo: remove everything but `analyzerConfigFiles` checks from here.